crates/server/src/admin/utils/auth.rs

```diff
@@ -292,3 +294,55 @@ pub fn validate_root_key_exists(
 
     Ok(req)
 }
+
+pub fn verify_near_public_key(public_key: &str, msg: &[u8], signature: &[u8]) -> EyreResult<bool> {
+    let public_key = bs58::decode(public_key)
+        .into_vec()
+        .map_err(|_| eyre!("Invalid public key: Base58 encoding error"))?;
+
+    let public_key = PublicKey::try_decode_protobuf(&public_key)
```

Comment:
why are we using protobuf?

Reply:
I don't remember :) It was long time ago, but it worked

Reply:
Understood, and left as-is, if I understand correctly, those pieces will be succeeded by the JWT flow.

Reply:
Not really. We are still using near as root key